### PR TITLE
Test for _id field that is not ObjectId

### PR DESCRIPTION
**Pull request by pcmiel**

### CODE REVIEW

File: src/test/java/com/futureprocessing/documentjuggler/integration/IdFieldTypesIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    private static final String COLLECTION_STRING_ID = "PersonsWithStringId";
+    private static final String COLLECTION = "Persons";
+    private static final String COLLECTION_OBJECT_ID_ID = "PersonsWithObjectIdId";
+
+    private static final String SURNAME = "Kowalski";
+
+    @CollectionName(COLLECTION)
+    interface Person {
+        @ObjectId
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/integration/IdFieldTypesIntegrationTest.java]: I think there should not be `@ObjectId` annotation when id is passed as `ObjectId` type.

**Comment by exys666** [comment]
+1

File: src/test/java/com/futureprocessing/documentjuggler/integration/IdFieldTypesIntegrationTest.java (hunk 1 above)
Comment (exys666) [inline src/test/java/com/futureprocessing/documentjuggler/integration/IdFieldTypesIntegrationTest.java]: You are right

File: src/test/java/com/futureprocessing/documentjuggler/integration/IdFieldTypesIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        //given
+        BaseRepository<Person> repository = new BaseRepository<>(db(), Person.class);
+        org.bson.types.ObjectId ID = new org.bson.types.ObjectId();
+
+        //when
+        repository.insert(p -> p.withId(ID).withSurname(SURNAME));
+
+        //then
+        BasicDBObject fromDb = (BasicDBObject) db().getCollection(COLLECTION).findOne();
+        assertThat(fromDb.getString("_id")).isEqualTo(ID.toHexString());
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/integration/IdFieldTypesIntegrationTest.java]: If you're trying to save _id as ObjectId it should be stored as it in database.

**Comment by szymonsobocik** [comment]
+1

**Pull request merged**